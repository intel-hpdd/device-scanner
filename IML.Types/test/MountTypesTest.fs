// Copyright (c) 2018 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

module IML.Types.MountTypesTest

open Fable.Import.Jest
open Matchers
open Thoth.Json

open IML.CommonLibrary

open MountTypes
open Fixtures

test "decode / encode Mounts" <| fun () ->
  fixtures.mounts
    |> Decode.decodeString LocalMounts.decoder
    |> Result.unwrap
    |> LocalMounts.encoder
    |> Encode.encode 2
    |> toMatchSnapshot
