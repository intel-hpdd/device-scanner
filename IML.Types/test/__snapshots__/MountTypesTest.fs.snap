// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode / encode Mounts 1`] = `
"[
  {
    \\"target\\": \\"/\\",
    \\"source\\": \\"/foo/bar\\",
    \\"fstype\\": \\"ext4\\",
    \\"opts\\": \\"ro\\"
  }
]"
`;
